browser/ui/views/split_view/split_view_location_bar.cc

```diff
@@ -0,0 +1,333 @@
+  flags.setAntiAlias(true);
+  flags.setStyle(cc::PaintFlags::kStroke_Style);
+  flags.setStrokeWidth(2);
+  canvas->DrawPath(path, flags);
+}
+
+gfx::Size SplitViewLocationBar::CalculatePreferredSize(
+    const views::SizeBounds& available_size) const {
+  auto size = views::WidgetDelegateView::CalculatePreferredSize(available_size);
+  if (auto* view = view_observation_.GetSource(); view && view->width()) {
+    size.set_width(std::min(view->width(), size.width()));
+  }
+  return size;
+}
+
+void SplitViewLocationBar::PrimaryPageChanged(content::Page& page) {
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::DidChangeVisibleSecurityState() {
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::WebContentsDestroyed() {
+  Observe(nullptr);
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::OnViewVisibilityChanged(views::View* observed_view,
+                                                   views::View* starting_view) {
+  UpdateVisibility();
+}
+
+void SplitViewLocationBar::OnViewBoundsChanged(views::View* observed_view) {
+  UpdateBounds();
+}
+
+void SplitViewLocationBar::OnViewIsDeleting(views::View* observed_view) {
+  view_observation_.Reset();
+}
+
+void SplitViewLocationBar::UpdateVisibility() {
+  auto* view = view_observation_.GetSource();
+  if (view && view->GetVisible()) {
+    GetWidget()->ShowInactive();
+  } else {
+    GetWidget()->Hide();
+  }
+}
+
+void SplitViewLocationBar::UpdateBounds() {
+  auto* view = view_observation_.GetSource();
+  if (!view) {
+    return;
+  }
+
+  gfx::Point point;
+  views::View::ConvertPointToWidget(view, &point);
+
+  auto* widget = GetWidget();
+  CHECK(widget);
+  GetWidget()->SetBounds(gfx::Rect(point, GetPreferredSize()));
+}
+
+void SplitViewLocationBar::UpdateURLAndIcon() {
+  if (auto* contents = web_contents()) {
+    auto format = GetURLFormatType();
+    if (HasCertError()) {
+      // In case of cert error, we show https scheme with another view so that
+      // we can strike the scheme. So we omit HTTPS here.
+      format |= url_formatter::kFormatUrlOmitHTTPS;
+      https_with_strike_->SetVisible(true);
+      scheme_separator_->SetVisible(true);
+    } else {
+      https_with_strike_->SetVisible(false);
+      scheme_separator_->SetVisible(false);
+    }
+
+    auto url = contents->GetLastCommittedURL();
+    auto origin = url::Origin::Create(url);
+    auto formatted_origin = url_formatter::FormatUrl(
+        origin.opaque() ? url : origin.GetURL(), format,
+        base::UnescapeRule::SPACES, nullptr, nullptr, nullptr);
+    url_->SetText(formatted_origin);
+    UpdateIcon();
+  } else {
+    https_with_strike_->SetVisible(false);
+    scheme_separator_->SetVisible(false);
+    url_->SetText({});
+    UpdateIcon();
+  }
+
+  UpdateBounds();
+}
+
+void SplitViewLocationBar::UpdateIcon() {
+  // At the moment, we show only warning icon.
+  safety_icon_->SetVisible(!IsContentsSafe());
+}
+
+bool SplitViewLocationBar::IsContentsSafe() const {
```

Comment:
there is a location bar model delegate to handle this stuff, like the actual url, we should not be writing our own logic for this stuff. This entire class is a major problem and I think we should revert this PR because we need a completely different approach here.